### PR TITLE
layers: Label missing pipeline index value

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Improve pipeline error messages

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1977,35 +1977,40 @@ bool CoreChecks::ValidatePipelineUnlocked(const PIPELINE_STATE *pPipeline, uint3
          IsDynamic(pPipeline, VK_DYNAMIC_STATE_DEPTH_COMPARE_OP_EXT) ||
          IsDynamic(pPipeline, VK_DYNAMIC_STATE_DEPTH_BOUNDS_TEST_ENABLE_EXT) ||
          IsDynamic(pPipeline, VK_DYNAMIC_STATE_STENCIL_TEST_ENABLE_EXT) || IsDynamic(pPipeline, VK_DYNAMIC_STATE_STENCIL_OP_EXT))) {
-        skip |=
-            LogError(device, "VUID-VkGraphicsPipelineCreateInfo-pDynamicStates-03378",
-                     "vkCreateGraphicsPipelines: Extended dynamic state used by the extendedDynamicState feature is not enabled");
+        skip |= LogError(device, "VUID-VkGraphicsPipelineCreateInfo-pDynamicStates-03378",
+                         "vkCreateGraphicsPipelines() pCreateInfos[%u]: Extended dynamic state used by the extendedDynamicState "
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:1981]: Should probably use `PRIu32` instead `%u` for consistency (here and elsewhere).

**Review by ncesario-lunarg — COMMENTED** [review]
Minor nit, but LGTM!

File: layers/core_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/core_validation.cpp:1981]: so every other spot in this function is using `%u`, I guess not sure when we started trying to use `PRIu32`... was there a reason for trying to switch to that?

If that is what we want, I rather just make everything in the function the same (`%u` or `PRIu32`)

Comment (jeremyg-lunarg) [reply layers/core_validation.cpp:1981]: Really it is up to gcc/clang -Wformat what you need to use. Personally I hate PRIxYZ but I hate printf format bugs even more.

Comment (ncesario-lunarg) [reply layers/core_validation.cpp:1981]: I think everything _should_ be using `PRI*` macros here, they just aren't. I would argue that new functionality should use the _slightly more type safe_ option (and maybe we'll actually refactor everything some day...).

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -1295,20 +1295,22 @@ bool CoreChecks::ValidatePipelineUnlocked(const PIPELINE_STATE *pPipeline, uint3
...
         if (subpass_desc && color_blend_state->attachmentCount != subpass_desc->colorAttachmentCount) {
-            skip |= LogError(
-                device, "VUID-VkGraphicsPipelineCreateInfo-attachmentCount-00746",
-                "vkCreateGraphicsPipelines() pCreateInfo[%u]: %s subpass %u has colorAttachmentCount of %u which doesn't "
-                "match the pColorBlendState->attachmentCount of %u.",
-                pipelineIndex, report_data->FormatHandle(pPipeline->rp_state->renderPass()).c_str(),
-                pPipeline->graphicsPipelineCI.subpass, subpass_desc->colorAttachmentCount, color_blend_state->attachmentCount);
+            skip |= LogError(device, "VUID-VkGraphicsPipelineCreateInfo-attachmentCount-00746",
+                             "vkCreateGraphicsPipelines() pCreateInfo[%" PRIu32
+                             "]: %s subpass %u has colorAttachmentCount of %u which doesn't "
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:1309]: So I think we are going to settle on the "concise and elegant" `PRI` macros as they are the safest option for now. Even though they make my eyes bleed a little...

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**